src/main/webapp/app/guided-tour/guided-tour.component.ts

```diff
@@ -173,68 +178,191 @@ export class GuidedTourComponent implements AfterViewInit, OnDestroy {
     /**
      * Scroll to and set highlighted element
      */
-    public scrollToAndSetElement(): void {
+    private scrollToAndSetElement(): void {
         this.selectedElementRect = this.updateStepLocation(this.getSelectedElement(), false);
         this.observeSelectedRectPosition();
 
         // Set timeout to allow things to render in order to scroll to the correct location
         setTimeout(() => {
-            if (this.isTourOnScreen()) {
+            if (this.isTourOnScreen(Direction.VERTICAL) && this.isTourOnScreen(Direction.HORIZONTAL)) {
                 return;
             }
-            const topPosition = this.getTopScrollingPosition();
-            try {
-                window.scrollTo({
-                    left: 0,
-                    top: topPosition,
-                    behavior: 'smooth',
-                });
-            } catch (err) {
-                if (err instanceof TypeError) {
-                    window.scroll(0, topPosition);
-                } else {
-                    throw err;
+            if (!this.isTourOnScreen(Direction.VERTICAL)) {
+                const topPosition = this.getTopScrollingPosition();
+                try {
+                    window.scrollTo({ left: 0, top: topPosition, behavior: 'smooth' });
+                } catch (err) {
+                    if (err instanceof TypeError) {
+                        window.scroll(0, topPosition);
+                    } else {
+                        throw err;
+                    }
                 }
             }
+            setTimeout(() => {
+                if (!this.isTourOnScreen(Direction.HORIZONTAL)) {
+                    this.flipOrientation();
+                }
+            }, 300);
         }, 0);
     }
 
+    /**
+     * Check if the current tour step has a bottom orientation
+     * @return true if the current tour step orientation is bottom, otherwise false
+     */
+    private isBottom(): boolean {
+        if (this.currentTourStep && this.currentTourStep.orientation) {
+            return (
+                this.currentTourStep.orientation === Orientation.BOTTOM ||
```

Comment:
You should be able to use just `this.orientation` here since you already assign it in the beginning

Reply:
Yes, but it refers to the orientation of the step, therefore it should refer to `this.currentTourStep.orientation`